File: misanuncios-ui/src/app/infrastructure/dto/roleHasPermission.dto.ts
import { PermissionDto } from "./permission.dto";
import { RoleDto } from "./role.dto";
import { RoleHasPermissionKeyDto } from "./roleHasPermissionKey.dto";

export interface RoleHasPermissionDto {

  id: RoleHasPermissionKeyDto;
  role: RoleDto; // Assuming Role model exists
  permission: PermissionDto; // Assuming Permission model exists

}

File: misanuncios/src/main/java/com/yaagoub/misanuncios/infrastructure/rest/spring/dto/RoleHasPermissionDto.java
package com.yaagoub.misanuncios.infrastructure.rest.spring.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(exclude = { "id","role","permission"})
public class RoleHasPermissionDto {

    private RoleHasPermissionKeyDto id;
    private RoleDto role;
    private PermissionDto permission;

}
